nvflare/app_common/job_schedulers/job_scheduler.py

```diff
@@ -116,6 +118,17 @@ def __init__(
         self.scheduled_jobs = []
         self.lock = threading.Lock()
 
+    def handle_event(self, event_type: str, fl_ctx: FLContext):
+        if event_type == EventType.JOB_STARTED:
+            with self.lock:
+                job_id = fl_ctx.get_prop(FLContextKey.CURRENT_JOB_ID)
+                self.scheduled_jobs.append(job_id)
```

Comment:
Should check against whether the job_id is already in self.scheduled_jobs, and append it only if the job_id is not in it.

Reply:
done